src/o365/graph/commands/teams/teams-channel-set.ts

```diff
@@ -0,0 +1,163 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import GraphCommand from '../../GraphCommand';
+import Utils from '../../../../Utils';
+import * as request from 'request-promise-native';
+import { Channel } from './Channel';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+interface Options extends GlobalOptions {
+  teamId: string;
+  channelName: string;
+  newChannelName?: string;
+  description?: string
+}
+class GraphTeamsChannelSetCommand extends GraphCommand{
+  public get name(): string {
+    return `${commands.TEAMS_CHANNEL_SET}`;
+  }
+  public get description(): string {
+    return 'Updates properties of a specified channel in the given Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.newChannelName = typeof args.options.newChannelName !== 'undefined';
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    return telemetryProps;
+  }
+
+   public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+    .ensureAccessToken(auth.service.resource, cmd, this.debug)
+    .then((): request.RequestPromise => {
+    const getrequestOptions: any = {
+      url: `${auth.service.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/channels?$filter=displayName eq '${encodeURIComponent(args.options.channelName as string)}'`,
```

Comment:
I don't think it's necessary to cast the channelName as string as it's already a string